content/blog/duckplyr-1-0-0/index.Rmd

```diff
@@ -0,0 +1,209 @@
+
+
+Then, the data manipulation pipeline uses the exact same syntax as a dplyr pipeline.
+The duckplyr package performs the computation using DuckDB.
+
+```{r}
+# Undo the effect of library(duckplyr)
+methods_restore()
+
+out <- babynames |>
+  as_duckdb_tibble() |>
+  mutate(prevalence = if_else(prop >= 0.01, "frequent", "rare")) |>
+  summarize(
+    .by = c(sex, year, prevalence),
+    babies_n = sum(n)
+  ) |>
+  filter(sex == "F")
+class(out)
+```
+
+In both cases, printing the result only shows the first few rows, as with dbplyr.
+
+```{r}
+out
+```
+
+The result can finally be materialized to memory, or computed temporarily, or computed to a file.
+
+```{r}
+# to memory
+nrow(out)
+# or for instance collect(out)
+
+# to a file
+csv_file <- withr::local_tempfile()
+compute_csv(out, csv_file)
+fs::file_size(csv_file)
+```
+
+When duckplyr itself does not support specific functionality, it falls back to dplyr.
+For instance, filtering on grouped data is not supported yet, still it works thanks to the fallback mechanism.
+
+```{r}
+babynames |>
+  filter(n > 10000, .by = "name")
+```
+
+For performance reasons, the output order of the result is not guaranteed to be stable.
+If you need a stable order, you can use `arrange()` or force output order stability by setting an environment variable.
+This and other limitations are documented in [`vignette("limits")`](https://duckplyr.tidyverse.org/articles/limits.html).
+
+## Large data
+
+For large data, duckplyr is a legitimate alternative to dtplyr and dbplyr.
+
+With large datasets, you want:
+
+- input data in an efficient format, like Parquet files, which duckplyr allows thanks to its ingestion functions like `read_parquet_duckdb()`.
+- efficient computation, which duckplyr provides via DuckDB's holistic optimization, without your having to use another syntax than dplyr.
+- the output to not clutter all the memory, which duckplyr supports through two features:
+    - computation to files using  [`compute_parquet()`](https://duckplyr.tidyverse.org/reference/compute_file.html) or [`compute_csv()`](https://duckplyr.tidyverse.org/reference/compute_file.html).
+    - the control of automatic materialization (collection of results into memory). You can disable automatic materialization completely or, as a compromise, disable it up to a certain output size. See [`vignette("prudence")`](https://duckplyr.tidyverse.org/articles/prudence.html) for details.
+
+See [`vignette("large")`](https://duckplyr.tidyverse.org/articles/large.html) for a walkthrough and more details.
+
+A drawback of analyzing large data with duckplyr is that the limitations of duckplyr won't be compensated by fallbacks, since fallbacks to dplyr necessitate putting data into memory.
+Therefore, if your pipeline encounters fallbacks, you might want to work around them by converting the duck frame into a table through `compute()` then running SQL code through the experimental `read_sql_duckdb()` function.
```

Comment:
I think it would be simpler to recommend running code to filter/sample/summarise to smaller than memory, then leave this advice to vignette.

Reply:
I'm removing this entirely for now.